domain/bankaccount.py

```diff
@@ -1,43 +1,68 @@
 from typing import List
-from domain.transaction import Transaction, create_transaction
-from domain.commontypes.types import AccountID
 
-Amount = float
+from domain.ledgeraccount import LedgerAccount
+from domain.transaction import Transaction, create_transaction
+from domain.commontypes.types import AccountID, Amount
 
 
 class BankAccount:
     def __init__(self,
                  account_id: AccountID,
-                 transactions: List[Transaction]
+                 main_account: LedgerAccount,
+                 draft_account: LedgerAccount,
+                 stage_account: LedgerAccount,
+                 overdraft_limit: Amount
                  ):
         self._id = account_id
-        self._transactions = transactions
+        self._main_account = main_account
+        self._draft_account = draft_account
+        self._stage_account = stage_account
+        self._overdraft_limit = overdraft_limit
 
     def get_balance(self) -> Amount:
-        balance = 0.0
-        for t in self._transactions:
-            if t.d_acc == self._id:
-                balance -= t.value
-            else:
-                balance += t.value
-        return balance
+        main_balance = self._main_account.get_balance()
+        draft_balance = self._draft_account.get_balance()
+        return main_balance - draft_balance
 
     def transfer(self, destiny_id: AccountID, value: Amount) -> None:
         if value <= 0:
             raise InvalidValueToTransfer(value)
 
-        balance = self.get_balance()
-        if balance < value:
-            raise InsufficientFundsException(balance, value)
+        main_balance = self._main_account.get_balance()
+        main_id = self._main_account.account_id
+        if main_balance < value:
+            remaining = value - main_balance
+            draft_account_balance = self._draft_account.get_balance()
+            overdraft = self._overdraft_limit - draft_account_balance
+
+            if overdraft >= remaining:
+                draft_id = self._draft_account.account_id
+                draft_transaction = create_transaction(draft_id, main_id, remaining)
+                main_transaction = create_transaction(main_id, destiny_id, value)
+
+                self._draft_account.add_transaction(draft_transaction)
+                self._main_account.add_transaction(draft_transaction)
+                self._main_account.add_transaction(main_transaction)
+
+            else:
+                raise InsufficientFundsException(self.get_balance(), value)
         else:
-            transaction = create_transaction(self._id, destiny_id, value)
-            self._transactions.insert(0, transaction)
+
+            transaction = create_transaction(main_id, destiny_id, value)
+            self._main_account.transactions.insert(0, transaction)
```

Comment:
why don't use the method to add transaction on main?

Reply:
thank you